Mathlib/AlgebraicGeometry/Sites/Small.lean

```diff
@@ -0,0 +1,254 @@
+      (fun Y f H j ↦ ((V f H).obj j).asOver S) (fun Y f H j ↦ ((V f H).map j).asOver S)
+    apply hV
+
+/-- The topology on `Over S` induced from the topology on `Scheme` defined by `P`.
+This agrees with the topology induced by `S.overPretopology P`, see
+`AlgebraicGeometry.Scheme.overGrothendieckTopology_eq_toGrothendieck_overPretopology`. -/
+abbrev overGrothendieckTopology : GrothendieckTopology (Over S) :=
+  (Scheme.grothendieckTopology P).over S
+
+lemma overGrothendieckTopology_eq_toGrothendieck_overPretopology :
+    S.overGrothendieckTopology P = (S.overPretopology P).toGrothendieck := by
+  ext X R
+  rw [GrothendieckTopology.mem_over_iff, Pretopology.mem_toGrothendieck]
+  constructor
+  · rintro ⟨T, ⟨𝒰, rfl⟩, hT⟩
+    letI (i : 𝒰.J) : (𝒰.obj i).Over S := { hom := 𝒰.map i ≫ X.hom }
+    letI : 𝒰.Over S :=
+      { over := inferInstance
+        isOver_map := fun i ↦ ⟨rfl⟩ }
+    use 𝒰.toPresieveOver, ⟨𝒰, inferInstance, rfl⟩
+    rwa [Cover.toPresieveOver_le_arrows_iff]
+  · rintro ⟨T, ⟨𝒰, h, rfl⟩, hT⟩
+    use Presieve.ofArrows 𝒰.obj 𝒰.map, ⟨𝒰, rfl⟩
+    rwa [Cover.toPresieveOver_le_arrows_iff] at hT
+
+variable {S}
+
+lemma mem_overGrothendieckTopology (X : Over S) (R : Sieve X) :
+    R ∈ S.overGrothendieckTopology P X ↔
+      ∃ (𝒰 : Cover.{u} P X.left) (_ : 𝒰.Over S), 𝒰.toPresieveOver ≤ R.arrows := by
+  rw [overGrothendieckTopology_eq_toGrothendieck_overPretopology]
+  constructor
+  · rintro ⟨T, ⟨𝒰, h, rfl⟩, hle⟩
+    use 𝒰, h
+  · rintro ⟨𝒰, h𝒰, hle⟩
+    exact ⟨𝒰.toPresieveOver, ⟨𝒰, h𝒰, rfl⟩, hle⟩
+
+variable [Q.IsStableUnderComposition]
+
+variable (S) {P Q} in
+lemma locallyCoverDense_of_le (hPQ : P ≤ Q) :
+    (MorphismProperty.Over.forget Q ⊤ S).LocallyCoverDense (overGrothendieckTopology P S) where
+  functorPushforward_functorPullback_mem X := by
+    intro ⟨T, hT⟩
+    rw [mem_overGrothendieckTopology] at hT ⊢
+    obtain ⟨𝒰, h, hle⟩ := hT
+    use 𝒰, h
+    rintro - - ⟨i⟩
+    have p : Q (𝒰.obj i ↘ S) := by
+      rw [← comp_over (𝒰.map i) S]
+      exact Q.comp_mem _ _ (hPQ _ <| 𝒰.map_prop i) X.prop
+    use (𝒰.obj i).asOverProp S p, MorphismProperty.Over.homMk (𝒰.map i) (comp_over (𝒰.map i) S), 𝟙 _
+    exact ⟨hle _ ⟨i⟩, rfl⟩
+
+instance : (MorphismProperty.Over.forget P ⊤ S).LocallyCoverDense (overGrothendieckTopology P S) :=
+  locallyCoverDense_of_le S le_rfl
+
+variable (S) {P Q} in
+/-- If `P` and `Q` are morphism properties with `P ≤ Q`, this is the Grothendieck topology
+induced via the forgetful functor `Q.Over ⊤ S ⥤ Over S` by the topology defined by `P`. -/
+abbrev smallGrothendieckTopologyOfLE (hPQ : P ≤ Q) : GrothendieckTopology (Q.Over ⊤ S) :=
+  letI : (MorphismProperty.Over.forget Q ⊤ S).LocallyCoverDense (overGrothendieckTopology P S) :=
+    locallyCoverDense_of_le S hPQ
+  (MorphismProperty.Over.forget Q ⊤ S).inducedTopology (S.overGrothendieckTopology P)
+
+/-- A special case of `smallGrothendieckTopologyOfLE` for the most common case, where `P = Q`. -/
```

Comment:
As this is the most important case, the docstring should be expanded so that we directly understand what it is.